connector_prestashop/models/account.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    prestashop_bind_ids = fields.One2many(
+        comodel_name='prestashop.refund',
+        inverse_name='odoo_id',
+        string='PrestaShop Bindings'
+    )
+
+    def action_move_create(self):
+        so_obj = self.env['prestashop.sale.order']
+        line_replacement = {}
+        for invoice in self:
+            sale_order = so_obj.search([('name', '=', invoice.origin)])
+            if not sale_order:
+                continue
+            sale_order = sale_order[0]
+            discount_product_id = sale_order.backend_id.discount_product_id.id
+            for invoice_line in invoice.invoice_line:
+                if invoice_line.product_id.id != discount_product_id:
+                    continue
+                amount = invoice_line.price_subtotal
+                partner = invoice.partner_id.commercial_partner_id
+                refund = self._find_refund(-1 * amount, partner)
+                if refund:
+                    invoice_line.unlink()
+                    line_replacement[invoice] = refund
+                    invoice.button_reset_taxes()
```

Comment:
In v8 it is: https://github.com/odoo/odoo/blob/75a196972e9e1056c8ab9c299ddfc8e2eaafd812/addons/account/account_invoice.py#L637